src/bindings/python/src/pyopenvino/core/common.hpp

```diff
@@ -80,22 +80,99 @@ std::vector<size_t> get_strides(const py::array& array);
 
 py::array as_contiguous(py::array& array, ov::element::Type type);
 
+template <typename T>
+py::array array_from_tensor_t(ov::Tensor&& t, py::dtype&& dtype) {
+    std::vector<T> tmp;
+    if (t.get_element_type() == ov::element::bf16) {
+        for (size_t i = 0; i < t.get_size(); i++) {
```

Comment:
can be converted to std::transform?
Get pointer to tensor before loop and use it.
`auto t_ptr = reinterpret_cast<const uint16_t*>(t.data());`
instead taking it every iteration.